### ISSUE TITLE
roachtest: Enable secondary tenants for mixed version/draining roachtests

### ISSUE THREAD

**Issue by rimadeodhar** [cockroachdb/cockroach#133165]
We should enable and test secondary tenant use case for mixed version and draining roachtests. This can ensure if the distributed sql related health checks added in 24.3 work as expected in the UA mode. 

This [slack thread](https://cockroachlabs.slack.com/archives/C06V95X6CVA/p1729615581678319) has more information on which tests to start with. This was identified as a 24.3 premortem action item from https://cockroachlabs.atlassian.net/wiki/spaces/DBSERV/pages/3985866818/DB+Server+24.3+Pre-mortem.

Jira issue: CRDB-43488

Epic CRDB-43487

**Comment by shubhamdhama** [comment]
Here are all the mixed-version tests,
```
backup-restore/mixed-version [disaster-recovery]
c2c/mixed-version [disaster-recovery]
cdc/mixed-versions [cdc]
change-replicas/mixed-version [kv]
declarative_schema_changer/job-compatibility-mixed-version-V242-V243 [sql-foundations]
decommission/mixed-versions [kv]
follower-reads/mixed-version/single-region [kv]
follower-reads/mixed-version/survival=region/locality=global/reads=strong [kv]
import/mixed-versions [sql-queries]
multi-region/mixed-version [test-eng]
rebalance/by-load/leases/mixed-version [kv]
rebalance/by-load/replicas/mixed-version [kv]
schemachange/mixed-versions [sql-foundations]
schemachange/mixed-versions-compat [sql-foundations]
sql-stats/mixed-version [obs-prs]
```
Which ones are relevant mixed-version tests for single-tenant mode so that we can later enable those for multi-tenant? TBD

In my opinion, “relevant” ones would have some nodes draining in a multi-version environment. If we don’t have such tests, should we (even) consider writing those? But even before we explore this option, is this exercise is to enable mixed-version/draining tests for secondary tenants in general or that tests this specific scenario.

**Comment by shubhamdhama** [comment]
There isn't any roachtest that would directly test the changes made in the linked PR but more importantly there isn't such use case that I think is not already covered by the unit tests. 

But at the time I didn't do the manual testing of actually upgrading a cluster from lower version to higher and see if the upgrade works. It was only tested via the unit tests. So today I spent some time doing this using a roachprod cluster of 5 nodes. Starting with 24.2 cluster version and upgrade only two nodes to 24.3 beta. Then different permutations of draining a single node while running distributed queries in a loop. Example, drain a node at 24.2 while having a 24.3 as the gateway and vice versa, all the binaries at 24.3 but auto finalization was disabled (i.e. cluster version is still 24.2).

I think we are good from the testing front.

Though while we are at it I tried enabling some tests in `roachtest/tests/drain.go` for secondary tenants as I think there aren't any roachtests that covers node drain for multitenancy. I was easily able to achieve this for one roachtest https://github.com/cockroachdb/cockroach/pull/133689 but hit the road block for the other which is reported here: #133692. Anyways, this will be taken care as a part of [CRDB-38970](https://cockroachlabs.atlassian.net/browse/CRDB-38970).
